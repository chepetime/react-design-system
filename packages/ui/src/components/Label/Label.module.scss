@use "./../../styles" as styles;
@use "~@chepe/tokens/tokens/tokens.scss" as tokens;

:root {
  --label-color: #000;
}

:global :root .dark {
  --label-color: #fff;
}

.Label {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  font-size: 1rem;
  margin-bottom: 1rem;
  display: block;

  color: var(--label-color);

  user-select: none;
  cursor: pointer;
}
