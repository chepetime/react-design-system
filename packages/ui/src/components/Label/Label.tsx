import React from "react";
import classnames from "classnames";
import css from "./Label.module.scss";

export interface LabelProps extends React.ComponentPropsWithRef<"label"> {
  disabled?: boolean;
  readOnly?: boolean;
  for?: string;
  htmlFor?: string;
  children?: React.ReactNode;
}

export const Label = React.forwardRef<HTMLLabelElement, LabelProps>(
  (
    { for: htmlFor = "", ...props }: LabelProps,
    ref: React.Ref<HTMLLabelElement>
  ) => {
    return (
      <label
        htmlFor={htmlFor}
        {...props}
        className={classnames(css.Label)}
        ref={ref}
      >
        {props.children}
      </label>
    );
  }
);

Label.displayName = "Label";
