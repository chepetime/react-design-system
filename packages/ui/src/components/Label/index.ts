export { Label } from "./Label";
export type { LabelProps } from "./Label";