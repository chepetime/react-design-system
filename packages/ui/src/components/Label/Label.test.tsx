import React from "react";
import { render } from "@testing-library/react";
import { Label } from "./Label";

describe("Label", () => {
  test("renders the Label component", () => {
    render(<Label />);
  });
});
