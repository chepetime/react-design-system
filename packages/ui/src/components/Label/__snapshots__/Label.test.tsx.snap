// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches snapshot 1`] = `
<div>
  <label
    class="Label"
    for=""
  />
</div>
`;
